src/app/Exec/lib/backend_proxy.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+const { BackendProxy: BackendProxyBase } = require("backend");
+const DirectBackend = require("./backends/direct/index");
+const JenkinsBackend = require("./backends/jenkins/index");
+const Backend = require("./types/backend");
+
+const BackendTypes = {
+    direct: DirectBackend,
+    jenkins: JenkinsBackend
+};
+
+class BackendProxy extends BackendProxyBase {
+    constructor() {
+        super(Backend);
+    }
+
+    createExecutor(backend, data = null) {
+        const backendType = this.getBackend(backend).backend.backendType;
+        if (Object.keys(this.executorClasses).indexOf(backendType) > -1) {
```

Comment:
I think this is easier to read:
`this.executorClasses.hasOwnProperty(backendType)`

Or this:
`backendType in this.executorClasses`